metricflow/sql/sql_plan.py

```diff
@@ -356,11 +358,11 @@ def render_node(self) -> SqlQueryPlanNode:  # noqa: D102
 class SqlCteNode(SqlQueryPlanNode):
     """Represents a single common table expression."""
 
-    select_statement: SqlSelectStatementNode
+    select_statement: SqlQueryPlanNode
```

Comment:
Notice this change got mixed in - this is to make recursive handling easier and it turns out the typed form is not necessary.